clients/client-py/test/test_utils.py

```diff
@@ -31,214 +33,188 @@ def dst(self, dt):
+        minute=1,
+        second=1,
+        tzinfo=utc
+    )
+    expected = '{"date":"2000-01-01T01:01:01Z"}'
+    actual = subject.dumpJson({'date': dateObj})
+    assert expected == actual
+
+
+def test_encode_string_for_b64_header():
+    # Really long strings trigger newlines every 72 ch
+    expected = 'YWJjZGVm' * 500
+    expected = expected.encode('ascii')
+    actual = subject.encodeStringForB64Header('abcdef' * 500)
+    assert expected == actual
+
+def test_makeb64urlsafe():
+    expected = b'-_'
+    actual = subject.makeB64UrlSafe('+/')
+    assert expected == actual
+
+def test_makeb64urlunsafe():
+    expected = b'+/'
+    actual = subject.makeB64UrlUnsafe('-_')
+    assert expected == actual
+
+
+def test_slug_id_is_always_nice():
+    with mock.patch('uuid.uuid4') as p:
+        # first bit of uuid set, which should get unset
+        p.return_value = uuid.UUID('bed97923-7616-4ec8-85ed-4b695f67ac2e')
+        expected = 'Ptl5I3YWTsiF7UtpX2esLg'
+        actual = subject.slugId()
+        assert expected == actual
+
+def test_slug_id_nice_stays_nice():
+    with mock.patch('uuid.uuid4') as p:
+        # first bit of uuid unset, should remain unset
+        p.return_value = uuid.UUID('3ed97923-7616-4ec8-85ed-4b695f67ac2e')
+        expected = 'Ptl5I3YWTsiF7UtpX2esLg'
+        actual = subject.slugId()
+        assert expected == actual
+
+
+def test_success_no_payload():
+    @httmock.all_requests
+    def response_content(url, request):
+        return {'status_code': 200, 'content': {}}
+
+    with httmock.HTTMock(response_content):
+        d = subject.makeSingleHttpRequest('GET', 'http://www.example.com', {}, {})
+        assert d.json() == {}
+        assert d.status_code == 200
+        d.raise_for_status()
+
+def test_success_payload():
+    @httmock.all_requests
+    def response_content(url, request):
+        assert request.body == 'i=j'
+        return {'status_code': 200, 'content': {'k': 'l'}}
+
+    with httmock.HTTMock(response_content):
+        d = subject.makeSingleHttpRequest('GET', 'http://www.example.com', {'i': 'j'}, {})
+        assert d.json() == {'k': 'l'}
+        assert d.status_code == 200
+        d.raise_for_status()
+
+def test_failure():
+    @httmock.all_requests
+    def response_content(url, requet):
+        return {'status_code': 404}
+
+    with httmock.HTTMock(response_content):
+        d = subject.makeSingleHttpRequest('GET', 'http://www.example.com', {}, {})
+        with pytest.raises(requests.exceptions.RequestException):
             d.raise_for_status()
 
-    def test_failure(self):
-        @httmock.all_requests
-        def response_content(url, requet):
-            return {'status_code': 404}
-
-        with httmock.HTTMock(response_content):
-            d = subject.makeSingleHttpRequest('GET', 'http://www.example.com', {}, {})
-            with self.assertRaises(requests.exceptions.RequestException):
-                d.raise_for_status()
 
+def test_success_put_file():
```

Comment:
@escapewindow 
This test case is failing with the error 
```
def test_success_put_file():
        with mock.patch.object(subject, 'makeSingleHttpRequest') as p:
            class FakeResp:
                status_code = 200
    
                def raise_for_status():
                    pass
    
            p.return_value = FakeResp()
>           subject.putFile('setup.py', 'http://www.example.com', 'text/plain')

test_utils.py:178:
```
---

```
filename = 'setup.py', url = 'http://www.example.com', contentType = 'text/plain'

    def putFile(filename, url, contentType):
>       with open(filename, 'rb') as f:
E       FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
taskcluster\utils.py:298: FileNotFoundError

Reply:
Looking.

Reply:
I was able to fix with:

```
diff --git a/clients/client-py/test/test_utils.py b/clients/client-py/test/test_utils.py
index a2bdda8e2..b20150064 100644
--- a/clients/client-py/test/test_utils.py
+++ b/clients/client-py/test/test_utils.py
@@ -171,11 +171,12 @@ def test_success_put_file():
         class FakeResp:
             status_code = 200

-            def raise_for_status():
+            def raise_for_status(*args):
                 pass

         p.return_value = FakeResp()
-        subject.putFile('setup.py', 'http://www.example.com', 'text/plain')
+        path = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'setup.py')
+        subject.putFile(path, 'http://www.example.com', 'text/plain')
         p.assert_called_once_with('put', 'http://www.example.com', mock.ANY, mock.ANY, mock.ANY)
```

The `raise_for_status` change avoids the error `TypeError: raise_for_status() takes 0 positional arguments but 1 was given` (it expects a `self` arg). What I'm doing with `os.path...` is finding the path of the test file (e.g., `client-py/test/test_utils.py`), finding the directory of the directory that holds it (e.g., `client.py/test/..`, which becomes `client-py/`), and then finding `setup.py` in that directory.